cmd/vela-terraform/plan.go

```diff
@@ -5,25 +5,178 @@
 package main
 
 import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
 	"github.com/sirupsen/logrus"
 )
 
 const planAction = "plan"
 
 // Plan represents the plugin configuration for plan information.
 type Plan struct {
+	// terraform file or directory to plan
+	Directory string
+	// If set, a plan will be generated to destroy all resources managed by the given configuration and state. i.e. "-destroy"
+	Destroy bool
+	// Return detailed exit codes when the command exits. i.e. "-detailed-exitcode"
+	DetailedExitCode bool
+	// ask for input for variables if not directly set. i.e. "-input=true"
+	Input bool
+	// the state file when locking is supported. i.e. -lock=true
+	Lock bool
+	// duration to retry a state lock. i.e. "-lock-timeout=0s"
+	LockTimeout time.Duration
+	// Specifies the depth of modules to show in the output. i.e. "-module-depth=n"
+	ModuleDepth int
+	// if specified, output won't contain any color. i.e. "-no-color"
+	NoColor bool
+	// Write a plan file to the given path. i.e. "-out=path"
+	Out string
+	// limit the number of parallel resource operations. i.e. "-parallelism=n"
+	Parallelism int
+	// update state prior to checking for differences. i.e. "-refresh=true"
+	Refresh bool
+	// path to read and save state (unless state-out is specified). i.e. "-state=path"
+	State string
+	// resource to target. i.e. "-target=resource"
+	Target string
+	// set a variable in the Terraform configuration. i.e. "-var 'foo=bar'"
+	Var []string
+	// set variables in the Terraform configuration from a file. i.e. "-var-file=foo"
+	VarFile string
```

Comment:
This should be type `[]string` since you can specify `-var-file` multiple times.

Reply:
Are you sure I'm not seeing that in the help docs? Specify it doesn't specify it can be used multiple times?

`terraform plan --help`
```sh
  -var 'foo=bar'      Set a variable in the Terraform configuration. This
                      flag can be set multiple times.

  -var-file=foo       Set variables in the Terraform configuration from
                      a file. If "terraform.tfvars" or any ".auto.tfvars"
                      files are present, they will be automatically loaded.
```

It does he plural in their docs online I just wasn't sure.
https://www.terraform.io/docs/commands/plan.html

> `-var-file=foo` - Set variables in the Terraform configuration from a variable file. If a terraform.tfvars or any .auto.tfvars files are present in the current directory, they will be automatically loaded. terraform.tfvars is loaded first and the .auto.tfvars files after in alphabetical order. Any files specified by -var-file override any values set automatically from files in the working directory. This flag can be used multiple times.

Reply:
You can also validate it locally with the files within https://gist.github.com/JordanSussman/e222e6690f744b76cc089209c376f9d7 by running the commands:

```sh
terraform init
terraform plan -var-file=vars1.json -var-file=vars2.json
```

Reply:
Oh perfect! Thanks, will make that change!